client/controllers/events/eventFiltration.coffee

```diff
@@ -60,3 +119,24 @@ Template.eventFiltration.events
     instance.$('.other-properties input:checked').each (i, input) ->
       otherProps[instance.removePropPrefix(input.id)] = true
     instance.properties.set(otherProps)
+
+  'change .locations select': (event, instance) ->
+    instance.countryLevel.set(event.target.value)
+
+  'click .location-list li': (event, instance) ->
+    selectedLocations = instance.selectedLocations
+    if selectedLocations.findOne(id: @_id)
```

Comment:
I changed the insert into upsert: https://github.com/ecohealthalliance/eidr-connect/pull/416/commits/10d195883d1470114afe63d1c701eab724438452#diff-91529dcc4459ea5068521029352eb3b6R69